src/Plugin/Field/FieldFormatter/CustomFormatters.php

```
@@ -133,28 +174,234 @@ public function viewElements(FieldItemListInterface $items, $langcode) {
+    $formatter_setting = $this->loadSettingEntity();
+    if (!$formatter_setting) {
+      return [];
+    }
+
+    $fields = $this->entityFieldManager->getFieldDefinitions('formatter_setting', (string) $formatter->id());
+    $configurable_fields = array_filter($fields, fn($f) => $f instanceof FieldConfigInterface);
+
+    $view_display = EntityViewDisplay::collectRenderDisplay($formatter_setting, 'default');
+
+    $settings = [];
+    foreach ($configurable_fields as $field_name => $field_definition) {
```

PHPMD (2.15.0)

[warning] Avoid unused local variables such as '$field_definition'. (undefined)

(UnusedLocalVariable)